### PR TITLE
Fix #4856 : Attributes from multiple scopes doesn't merge

### LINKED ISSUE #4856: Scopes attributes are not merged
Version : **3.13.0**
Dialect : **mysql**

**Given**
 
```js
var User = sequelize.define('User', {
    username: {
        type: DataTypes.STRING,
        allowNull: false
    },
    about: {
        type: DataTypes.STRING
    },
    password: {
        type: DataTypes.STRING,
        allowNull: false,
        set: function (val) {
            // Some encrypting logic
        }
    }
}, {
    scopes: {
        'public': {
            attributes: [
                'id',
                'username',
                'about'
            ]
        },
        'private': {
            attributes: [
                'password'
            ]
        }
    }
});
```

**When** querying an instance with multiple scopes, scopes attributes do not get merged. 
e.g.

```js
User.scope('public','private').findOneById(1).then(function(user){
    console.log(user.about); // Exists, attribute included from public scope
    console.log(user.password); // Undefined, public and private attributes were not merged
});
```

### PR DESCRIPTION
**Pull request by sushantdhiman**
Fixes #4856

### CODE REVIEW

File: test/integration/model/scope.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                expect(record.username).to.exist;
+                expect(record.access_level).to.exist;
+              });
+    });
+    it('should be able to merge attributes as object', function () {
+      return this.ScopeMe.scope('lowAccessObj','withNameObj').findOne()
+              .then(function(record){
+                expect(record.other_value).to.exist;
+                expect(record.username).to.exist;
+              });
```
Comment (mickhansen) [inline test/integration/model/scope.test.js]: what about `access_level` in this case?

Comment (sushantdhiman) [inline test/integration/model/scope.test.js]: `access_level` is removed because scope `withNameObj` have `excluded` it.

Comment (mickhansen) [inline test/integration/model/scope.test.js]: please include that in the test then :)

**Comment by mickhansen** [comment]
Will need a rebase aswell.

**Comment by sushantdhiman** [comment]
It appears to be including the `access_level` apparently due to its presence in one of the scopes. What should I do in such case? Remove it or keep it?

**Comment by sushantdhiman** [comment]
Here is the implementation details @mickhansen @janmeier , Need your comments.

When dealing with ambiguous `include` and `exclude` parameters. We have two choices

* Give preference to `exclude` parameters
* Give preference to `include` parameters

Current PR is designed to give preference to `include` parameters. This means if one scope try to `include` and attribute and other scope try to `exclude` it, we will choose to `include` it in final result.

Let me know what do you think about it :)

**Comment by mickhansen** [comment]
@sushantdhiman Can we start with a simple implementation that simply merges attributes when they are arrays? That should solve the immediate problem for most users, after that we can come up with a design for handling include/exclude.

**Comment by sushantdhiman** [comment]
@mickhansen , it solves that problem already

**Comment by mickhansen** [comment]
@sushantdhiman This PR already solves it or it's already solved? If this PR solves it just change the tests/implementation to do pure array concat implementation for now.

**Comment by sushantdhiman** [comment]
@mickhansen Done

**Pull request merged**